src/Donors/ServiceProvider.php

```diff
@@ -65,4 +69,22 @@ private function addCustomFieldsToDonorDetails()
             echo (new DonorDetailsController())->show($donor);
         });
     }
+
+    /**
+     * Hook into the donor creation process to ensure that donors are also users.
+     * @unreleased
+     */
+    protected function enforceDonorsAsUsers()
+    {
+        add_action('givewp_donate_controller_donor_created', function(Donor $donor, $formId) {
+            if(!$donor->userId) {
+
+                give(CreateUserFromDonor::class)->__invoke($donor);
```

Comment:
Let's add that try catch here and at least log any errors.

Reply:
Resolved, see https://github.com/impress-org/givewp/pull/7103/commits/b83c387611ae7a4ac8ef88e5e61070681d972566.